### ISSUE TITLE
Exporter not working unless scenarios are defined in the scenario tree

### ISSUE THREAD

**Issue by FebinKa** [spine-tools/Spine-Toolbox#2248]
I have model solutions for SpineOpt from different scenarios but the exporter does not export one of them. "Run SpineOpt.." was only defined using the alternative tree before I added the other alternatives and scenarios and I could not get the Exporter to save the results from this case. However, it worked for all other iterations, including "Base__Run SpineOpt..", which is the same model but initialized in the Scenario tree.

![SpineExporter](https://github.com/spine-tools/Spine-Toolbox/assets/55534006/607bc45b-be58-45aa-9da5-534cf668be89)
![SpineExporter2](https://github.com/spine-tools/Spine-Toolbox/assets/55534006/a043887d-cb50-44f4-b8c6-f89a55c94f10)

**Comment by jkiviluo** [comment]
I see three scenarios in the scenario tree and all of those are also visible in the export preview. So, not sure what the problem really is. 

The 'scenarios' turn into 'alternative' results after you have executed SpineOpt (although at the same time, also a scenario is created for the new alternative containing scenario results). Alternatives as such are not run in SpineOpt (nor are they processed in workflows - it's always scenarios that progress through workflow). You need to make a scenario with only that alternative, if you want that alternative to be run by SpineOpt.

**Comment by nnhjy** [comment]
> So in this case where I have an initial run from an alternative and then introduce scenarios, the only way to save the results from the alternative would be by re-running the initial alternative again as a scenario even though I have the results already?

Yes if you would like to have a new result of the initial alternative along with those of the introduced scenarios. In that case, the filter of the connection between the input database and the SpineOpt tool (the hammer) is activated, selecting the execution as per the scenario.

**Comment by nnhjy** [comment]
> I see three scenarios in the scenario tree and all of those are also visible in the export preview. So, not sure what the problem really is.

Correct me if I am wrong @FebinKa. The problem is the exporter overlooks results generated by either the `Base` alternative or `Base` scenario. It looks weird in that the results are only stored as per alternative in the output database regardless of how the SpineOpt run is defined (by a single alternative or scenarios) in the input database.

**Comment by jkiviluo** [comment]
Base scenario in the input database should turn into Base alternative (with timestamp and toolstamp) in the output DB (provided that the scenario is run - it needs to be checked in the arrow filter as @nnhjy is saying). There will then also be Base scenario in the output database that contains all historical runs for the Base scenario (as alternatives).
